alf/algorithms/monet_algorithm.py

```diff
@@ -90,42 +90,43 @@ def __init__(self,
             input_tensor_spec=input_tensor_spec).output_spec
 
         self._nonskip_mlp = alf.networks.EncodingNetwork(
-            input_tensor_spec=last_skip_tensor_spec,
-            input_preprocessors=alf.layers.Reshape(-1),
-            fc_layer_params=nonskip_fc_layers,
-            last_layer_size=last_skip_tensor_spec.numel,
-            last_activation=torch.relu_)
+            input_tensor_spec=alf.TensorSpec((last_skip_tensor_spec.numel, )),
+            fc_layer_params=nonskip_fc_layers)
 
-        self._reshape = alf.layers.Reshape(*last_skip_tensor_spec.shape)
+        self._reshape = torch.nn.Sequential(
+            alf.layers.FC(nonskip_fc_layers[-1], last_skip_tensor_spec.numel,
+                          torch.relu_),
```

Comment:
This can be part of nonskip_mlp. And should update document of nonskip_fc_layers about the additional FC layer.

Reply:
> This can be part of nonskip_mlp. And should update document of nonskip_fc_layers about the additional FC layer.

The reason why I chose to put this additional FC layer in `self._reshape` instead of `self._nonskip_mlp` is that sometimes we want to perform extra operations on the output of `self._nonskip_mlp` (as in the vonet case). This output can be obtained by calling `self.encode()`. If the additional FC is in `self._nonskip_mlp` then the encoding dim might be large depending on `last_skip_tensor_spec.numel` and is not straightforwardly controllable.

Reply:
> Should still comment about this additional fc in doc-string of the class.

added